mainsite/oauth2/validators.py

```diff
@@ -30,3 +35,80 @@ def get_additional_claims(self):
             ("verified_user", claims.UserVerified([SupportedScopes.PROFILE])),
             ("networks", claims.Networks([SupportedScopes.NETWORKS])),
         ]
+
+    def invalidate_authorization_code(self, client_id, code, request, *args, **kwargs):
+        print("invalidate_authorization_code")
+        grant = Grant.objects.get(code=code, application=request.client)
+        self.grant = grant
+        return super().invalidate_authorization_code(
+            client_id, code, request, *args, **kwargs
+        )
+
+    def _create_access_token(self, expires, request, token, source_refresh_token=None):
```

Comment:
Does this mean the `amr` won't be included in the JWT ID token?

The JWT ID token is sometimes used by the client (e.g. single-page application) to present personal information and when not reaching out to other OAuth2 resource servers but the client is a confidential one, it is all the info the application will consume so I would still add it to the JWT ID token too if it is not what this method does.

Reply:
@caguado added to JWT ID token with https://github.com/peeringdb/peeringdb/pull/1565/commits/e27ab9dce69c8036262b27322a190a4f02889234